README.md

```diff
@@ -1 +1,1148 @@
+[Build]   [main]  ---> 392c3f6339f7
+[Build]   [main] Step 2/9 : WORKDIR /usr/src/app
+[Build]   [main]  ---> Using cache
+[Build]   [main]  ---> c27a2f22ba27
+[Build]   [main] Step 3/9 : COPY package.json package-lock.json ./
+[Build]   [main]  ---> Using cache
+[Build]   [main]  ---> fc79948ab18b
+[Build]   [main] Step 4/9 : RUN npm install --ci --production     && npm cache clean --force     && rm -f /tmp/*
+[Build]   [main]  ---> Using cache
+[Build]   [main]  ---> 0cdc6d1d7af9
+[Build]   [main] Step 5/9 : COPY . ./
+[Build]   [main]  ---> fec02483c800
+[Build]   [main] Step 6/9 : RUN echo "Rebuild the image"
+[Build]   [main]  ---> Running in 0c1f2cbca19f
+[Build]   [main] Rebuild the image
+[Build]   [main] Removing intermediate container 0c1f2cbca19f
+[Build]   [main]  ---> aec92835242c
+[Build]   [main] Step 7/9 : CMD ["npm", "start"]
+[Build]   [main]  ---> Running in 824e56df2920
+[Build]   [main] Removing intermediate container 824e56df2920
+[Build]   [main]  ---> c49d013039de
+[Build]   [main] Step 8/9 : LABEL io.resin.local.image=1
+[Build]   [main]  ---> Running in e9158ba83571
+[Build]   [main] Removing intermediate container e9158ba83571
+[Build]   [main]  ---> 2317085d9161
+[Build]   [main] Step 9/9 : LABEL io.resin.local.service=main
+[Build]   [main]  ---> Running in ebcaf1a3351d
+[Build]   [main] Removing intermediate container ebcaf1a3351d
+[Build]   [main]  ---> fc5e4459c406
+[Build]   [main] Successfully built fc5e4459c406
+[Build]   [main] Successfully tagged local_image_main:latest
+[Logs]    [9/5/2019, 3:58:56 PM] [main]
+[Logs]    [9/5/2019, 3:58:56 PM] [main] > balena-cli-masterclass@1.0.0 start /usr/src/app
+[Logs]    [9/5/2019, 3:58:56 PM] [main] > node src/helloworld.js; sleep infinity
+[Logs]    [9/5/2019, 3:58:56 PM] [main]
+[Logs]    [9/5/2019, 3:58:56 PM] [main] Hello world!
+```
+Once rebuilt, it will restart the service. Notice that the `Rebuild the image`
+echoed line is now in the build log.
+
+Livepush goes way further than this, however. Only files that affect the
+building of the service force a rebuild. For other files, for example source
+files that run in-service, the Supervisor replaces the files in-situ in the
+relevant container layer.
+To show this, continue to run the `balena push` command and then alter the
+`src/helloworld.js` in a text editor and change:
+```
+console.log('Hello world!');
+```
+to
+```
+console.log('Hello moon!');
+```
+On saving the file, you'll see the following output:
+```
+[Live]    Detected changes for container main, updating...
+[Live]    [main] Restarting service...
+[Logs]    [9/5/2019, 4:02:27 PM] [main]
+[Logs]    [9/5/2019, 4:02:27 PM] [main] > balena-cli-masterclass@1.0.0 start /usr/src/app
+[Logs]    [9/5/2019, 4:02:27 PM] [main] > node src/helloworld.js; sleep infinity
+[Logs]    [9/5/2019, 4:02:27 PM] [main]
+[Logs]    [9/5/2019, 4:02:28 PM] [main] Hello moon!
+```
+Instead of rebuilding the image, which takes time, the file is injected into
```

Comment:
Correct me if I am mistaken here, but this change also forces all subsequent layers to be rebuilt as well right?

Reply:
Nope, it just goes into a new layer on top AFAIK. It simply dumps it in, then restarts the container, nothing gets rebuilt.